osu.Server.Spectator/Hubs/ServerMultiplayerRoom.cs

```diff
@@ -165,13 +141,15 @@ async Task start()
                         if (roomUsage.Item == null)
                             return;
 
-                        roomUsage.Item.Countdown = null;
-
-                        await hub.NotifyNewMatchEvent(roomUsage.Item, new CountdownChangedEvent { Countdown = null });
+                        // A new countdown may have since replaced the currently tracked countdown.
+                        if (roomUsage.Item.Countdown != countdown)
+                            return;
```

Comment:
As far as I can tell, this is a newly-added check, however I logically can't follow the case where this would be required. Because any time where `StopCountdown()` is called, we must already be inside a usage of a room, which means `stopSource` should always be cancelled before this code is run and the check following this one should enough to cover this case.

Am I missing something?

Reply:
I think it may no longer be required. Previously I didn't have the `StopCountdown` flow synchronous (as it is when I pushed the latest changes) and caused this requirement. Will re-test.

Reply:
Seems to work fine without it. I've added an assert in place as I think it makes it more obvious as to expectations.